apps/testingaccessibility/src/server/ability.ts

```diff
@@ -35,6 +35,11 @@ function defineAbilityFor(token: any) {
 
   if (hasValidPurchase(token.purchases)) {
     can('view', 'Content')
+    can('view', 'Product', {
+      productId: {
+        $in: token.purchases.map((purchase: Purchase) => purchase.productId),
+      },
+    })
```

Comment:
checks if a specific productId is in a list of purchases

CASL uses this mongoDB query language for advanced querying of the "subject" which offers a lot of flexibility https://casl.js.org/v5/en/guide/conditions-in-depth#mongo-db-and-its-query-language